File: services/shop-backend/src/main/java/io/miragon/shop/adapter/outbound/persistence/article/ArticleJpaRepository.java
package io.miragon.shop.adapter.outbound.persistence.article;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface ArticleJpaRepository extends JpaRepository<ArticleEntity, UUID> {
    List<ArticleEntity> findAllByIdIn(List<UUID> ids);
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/adapter/outbound/persistence/article/ArticleJpaRepository.kt
package io.miragon.shop.adapter.outbound.persistence.article

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ArticleJpaRepository : JpaRepository<ArticleEntity, UUID> {
    fun findAllByIdIn(ids: List<UUID>): List<ArticleEntity>
}